_posts/2021-02-12-generalist-specialization-and-the-impostor-syndrome.md

```diff
@@ -0,0 +1,80 @@
+
+Neglecting any of these, in a professionally done project, can doom it to failure, or at least make it very difficult. So to be able to do your work properly, there's an endless stream of things you need to be aware of and make the right choices. The more you know the more you know what you don't know.
+
+Furthermore, things move rapidly. While you can use your knowledge from decades back, in our line of work we typically need to be constantly learning more, keeping skills up to date, replacing old tools with new, better ones. While at some point in time you could say you know everything there is about programming, now things have exploded exponentially. If anyone tells you they know all there is to know about software delivery, they are lying.
+
+So, that's the scene. That's the setup. Things might be a bit calmer if you're working on the same product/service for years, but I can tell you that especially in consulting business, you have to push yourself to new areas constantly, and quickly master them. Many people actually enjoy that, on a good day it's a rush to be able to grow and learn new things.
+
+But then there's that moment of self-doubt and exhaustion. Did I over-reach? Did I promise too much?
+
+## What's a generalist specialization?
+
+In our line of work, some amount of specialization is a must, it's also a natural thing. Without specializing, without spending a bit of time with some technology or method, it's hard to master it. Mastering gives a satisfying feeling, whether it be programming languages like Java, Clojure, TypeScript, Kotlin, or cloud platforms like AWS.
+
+The thing is, especially in the software consulting line of business, doubling down on just one thing won't get you very far. It might not even carry you through the first job interview unless you are applying for very entry-level work with some training coming up immediately. That's because we don't know what skills will be needed for the next project. If you are only a master of one skill, it might be more difficult to find yourself a good project to work with, since others in the team would need to carry all other areas and aspects of the project, until you have learned them. It gives a much better base for consulting work to be good with many parts of the software craft (as opposed to being a master of one).
+
+So, while specialization is still rather a valid thing to do, you should also invest in learning a wide array of skills for software crafting, as listed above. These days you definitely need to understand automated testing, delivery, security, and cloud platforms, in addition to Frontend, Backend, and databases. But you can still specialize, you can double on on one or two things that bring you pleasure. You can actively practice them, learn more about them, and be the go-to-person for said techniques.
```

Comment:
Listed where? Do you mean the list of programming languages and AWS?

No need to capitalize frontend and backend.

"double down on"